osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/processors/KeysRequestMessageProcessor.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2015 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.protocol.dlms.infra.messaging.processors;
+
+import java.io.Serializable;
+import java.time.Duration;
+import javax.jms.JMSException;
+import javax.jms.ObjectMessage;
+import lombok.extern.slf4j.Slf4j;
+import org.opensmartgridplatform.adapter.protocol.dlms.application.services.ConfigurationService;
+import org.opensmartgridplatform.adapter.protocol.dlms.application.services.DeviceKeyProcessingService;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.entities.DlmsDevice;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.factories.DlmsConnectionManager;
+import org.opensmartgridplatform.adapter.protocol.dlms.exceptions.DeviceKeyProcessAlreadyRunningException;
+import org.opensmartgridplatform.adapter.protocol.dlms.infra.messaging.DeviceRequestMessageProcessor;
+import org.opensmartgridplatform.adapter.protocol.dlms.infra.messaging.DeviceRequestMessageSender;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.SetKeysRequestDto;
+import org.opensmartgridplatform.shared.exceptionhandling.OsgpException;
+import org.opensmartgridplatform.shared.infra.jms.MessageMetadata;
+import org.opensmartgridplatform.shared.infra.jms.MessageType;
+import org.springframework.beans.factory.annotation.Autowired;
+
+@Slf4j
+public class KeysRequestMessageProcessor extends DeviceRequestMessageProcessor {
+
+  @Autowired private ConfigurationService configurationService;
+
+  @Autowired private DeviceRequestMessageSender deviceRequestMessageSender;
+
+  @Autowired private DeviceKeyProcessingService deviceKeyProcessingService;
+
+  public KeysRequestMessageProcessor(final MessageType messageType) {
+    super(messageType);
+  }
+
+  @Override
+  protected Serializable handleMessage(
+      final DlmsConnectionManager conn,
+      final DlmsDevice device,
+      final Serializable requestObject,
+      final MessageMetadata messageMetadata)
+      throws OsgpException {
+
+    this.assertRequestObjectType(SetKeysRequestDto.class, requestObject);
+
+    final SetKeysRequestDto keySet = (SetKeysRequestDto) requestObject;
+    this.configurationService.replaceKeys(conn, device, keySet, messageMetadata);
+
+    return null;
+  }
```

Comment:
The `KeysRequestMessageProcessor` should not do this. The `SetKeysRequestDto` that is asserted and acted upon, is the type for only one specific processor (the `ReplaceKeysRequestMessageProcessor`).

Reply:
Indeed method removed